src/nnet3/nnet-normalize-component.h

```diff
@@ -269,6 +272,28 @@ class BatchNormComponent: public Component {
   // (smaller -> slower).  The same config exists in NormalizeComponent.
   BaseFloat target_rms_;
 
+  // the flag whether use the batch renorm described in paper:
+  // https://arxiv.org/abs/1702.03275
+  // This adds extra variables during training. The inference is the same 
+  // for either value of this parameter.
+  bool batch_renorm_;
```

Comment:
I would prefer you to do this as a separate component, it will keep the code cleaner.
Of course, you'd have to modify code about setting batchnorm test-mode, to do this type of component too.